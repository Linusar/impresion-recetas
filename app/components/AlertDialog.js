import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import { ModalContext } from '../context/ModalContext';

const propTypes = {
  title: PropTypes.string,
  children: PropTypes.node
};

const AlertDialog = ({ children, title }) => {
  const { modal, setModal } = useContext(ModalContext);

  function handleClose() {
    setModal(false);
  }

  return (
    <React.Fragment>
      <Dialog
        open={modal}
        onClose={handleClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
        <DialogContent>{children}</DialogContent>
        <DialogActions>
          <Button
            onClick={handleClose}
            color="primary"
            autoFocus
            data-test="alertDialogButtonClose"
          >
            Cerrar
          </Button>
        </DialogActions>
      </Dialog>
    </React.Fragment>
  );
};

AlertDialog.defaultProps = {
  title: 'Im a cool modal',
  children: <div />
};

AlertDialog.propTypes = propTypes;

export default AlertDialog;
