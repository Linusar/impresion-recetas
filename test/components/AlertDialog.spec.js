import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Button from '@material-ui/core/Button';
import ModalContextProvider from '../../app/context/ModalContext';
import AlertDialog from '../../app/components/AlertDialog';

Enzyme.configure({ adapter: new Adapter() });

const TestComponent = () => (
  <ModalContextProvider value={{ modal: true }}>
    <AlertDialog />
  </ModalContextProvider>
);

const setup = () => {
  const component = shallow(<TestComponent />);
  return {
    component,
    buttons: component.find(Button)
  };
};

describe('Alert Dialog render', () => {
  it('should render AlertDialog', () => {
    const { component } = setup();
    expect(component.find(AlertDialog).exists()).toBe(true);
  });

  it('should match exact snapshot', () => {
    const dialog = (
      <div>
        <Router>
          <TestComponent />
        </Router>
      </div>
    );
    const tree = renderer.create(dialog).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
