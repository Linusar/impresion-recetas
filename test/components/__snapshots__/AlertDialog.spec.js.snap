// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Dialog render should match exact snapshot 1`] = `<div />`;
